### PR TITLE
Avoid noise when code runs with Ruby warnings

### PR DESCRIPTION
**Pull request by cpetschnig**
New versions of RSpec have warnings turned on by default:

    # This setting enables warnings. It's recommended, but in some cases may
    # be too noisy due to issues in dependencies.
    config.warnings = true

I believe these warnings are helpful in general to write better code. However, Grape Entity is very noisy. I get more than 10 lines of warnings for one single test where Grape Entity is involved.

I hope you agree. Some of my changes could also be solved differently for sure. I am happy to discuss and change those cases. I also turned warnings on in the test suite.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 #### Features
 
-* Your contrbution here.
+* Your contribution here.
```
Comment (LeFnord) [inline CHANGELOG.md]: thanks

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
 
 #### Fixes
 
-* Your contrbution here.
```
Comment (LeFnord) [inline CHANGELOG.md]: please re-add it below your contribution

**Comment by LeFnord** [comment]
thanks @cpetschnig for contributing, please see my comment above

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -7,4 +7,8 @@
 
 Bundler.require :default, :test
 
-RSpec.configure(&:raise_errors_for_deprecations!)
+RSpec.configure do |config|
+  config.raise_errors_for_deprecations!
+
+  config.warnings = true
+end
```
Comment (LeFnord) [inline spec/spec_helper.rb]: please revert this change

Comment (LeFnord) [inline spec/spec_helper.rb]: @dblock this would enable the warnings and makes also the specs noisy, maybe they could be used for writing better code, but think for that we have rubocop, so by enabling it, we have two of such tools … mmh

if they both give the same or similar warnings, then one is enough and the rubocop messages much more helpful, IMHO the rspec warnings are higher level and helpful for experienced devs, but they also raises the barrier for contributions

if not then it should be disabled

Comment (dblock) [inline spec/spec_helper.rb]: I agree with @LeFnord.

**Comment by cpetschnig** [comment]
Thank you for your feedback. I have updated the PR.

**Comment by LeFnord** [comment]
👍
